### ISSUE TITLE
PSD2 - Onestepcheckout and redirect issue

### ISSUE THREAD

**Issue by MarcoPierozziTriboo** [Adyen/adyen-magento#1040]
**Magento version**: 1.9.4.1
**Plugin version**: 2.15.1
**Description**

When Enabling 3DS2 on onestepcheckout the adyen/payment/threeds2-js-utils.js is missing from the layout file, so no extra info are loaded for credit cards and 3ds2 auth fails

Fixed by adding in adyen.xml layout file
```xml
<onestepcheckout_index_index>
    <reference name="head">
        <action method="addJs">
            <script>adyen/payment/threeds2-js-utils.js</script>
        </action>
    </reference>
</onestepcheckout_index_index>
```

Beside that in app/design/frontend/base/default/template/adyen/threeds2.phtml in the final part of the script the redirect urls are hardcoded in the script (example `window.location = '/checkout/cart'`). If you have store views in url, this breaks redirect logic. A more correct approach is:
```javascript
promise.then(function res(result) {
    if (result) {
        var response = JSON.parse(result);
        if (response.authentication) {
            renderThreeDS2Component(response["resultCode"], response["authentication"]["threeds2.challengeToken"]);
        } else if (response === "Authorised") {
                window.location = "<?php echo $this->getUrl('checkout/onepage/success'); ?>";
        } else if (response === "Refused" || response === "Error") {
                window.location = "<?php echo $this->getUrl('checkout/cart'); ?>";
        }
    } else {
        window.location = "<?php echo $this->getUrl('checkout/cart'); ?>";
    }
});
```

**Comment by Aleffio** [comment]
Hi @MarcoPierozziTriboo ,

Indeed we need to add the store views to the url,
could you please open a PR to our repo with your suggestion?

Thanks,
Alessio
Adyen

**Comment by Aleffio** [comment]
Fixed in #1047 ,
closing the issue.

Regards,
Alessio
Adyen
